projects/igniteui-angular/src/lib/directives/text-highlight/text-highlight.directive.ts

```diff
@@ -179,6 +183,26 @@ export class IgxTextHighlightDirective implements AfterViewInit, AfterViewChecke
     @DeprecateProperty(`IgxTextHighlightDirective 'page' input property is deprecated.`)
     public page: number;
 
+    /**
+     * A map that contains all aditional checks, that you need, in order to activate
+     * your highlighted element. To activate the check, you will have to add a new metadata key to
+     * the `metadata` property into the IActiveHighlightInfo interface.
+     *
+     * @example
+     * ```typescript
+     * // Set a property, which would disable the highlight for a given element on a cetain condition
+     * if (dontActivateCondition) {
+     *     cell.highlight.metadata.set('highlightElement', false);
+     * }
+     *
+     * const metadataMap = new Map<string, any>();
+     * metadataMap.set('highlightElement') = true;
+     * IgxTextHighlightDirective.setActiveHighlight(id, {index: index, metadata: metadataMap});
```

Comment:
I don't think this code snippet is appropriate since it is cell specific. We should have something similar to the row and column snippets.

Reply:
I have updated it, although I'm not sure if it's as readable as it should be...